vpr/src/route/spatial_route_tree_lookup.cpp

```diff
@@ -33,7 +33,7 @@ void update_route_tree_spatial_lookup_recur(t_rt_node* rt_node, SpatialRouteTree
     auto& device_ctx = g_vpr_ctx.device();
     const auto& rr_graph = device_ctx.rr_graph;
 
-    auto& rr_node = device_ctx.rr_nodes[rt_node->inode];
+    auto& rr_node = rr_graph.rr_nodes()[rt_node->inode];
```

Comment:
Since we have finished all the APIs, you can consider to replace this line by 

```
RRNodeId rr_node = rt_node->inode;
```
 And also adapt the rest codes in this function, ``rr_node.id()`` -> ``rr_node``